services/embed_docsite/github_utils.py

```diff
@@ -0,0 +1,101 @@
+import requests
+from util import create_logger
+
+logger = create_logger("GitHubUtils")
+
+def download_main_docs(github_info):
+    """Downloads and processes the main docs from GitHub API info."""
+    output = []
+
+    for file_info in github_info:
+        try:
+            response = requests.get(file_info["download_url"])
+            response.raise_for_status()
+            
+            output.append({
+                "name": file_info["name"],
+                "docs": response.text
+            })
+
+        except requests.RequestException as e:
+            logger.info(f"Failed to fetch content for {file_info['name']}: {e}")
+    logger.info(f"Downloaded and processed {len(output)} files from GitHub")
+    logger.info(f'{output[0]}')
+    return output
+
+def get_github_urls(repo, path="", owner="OpenFn", file_type=".md"):
+    """"
+    Get the download URLs for a GitHub repository.
+
+    :param repo: The repository (e.g. "docs")
+    :param path: The path from root (e.g. "")
+    :param owner: The repository owner (default="OpenFn")
+    :return: List of dictionaries {name, path, download_url}
+    """
+    files = []
+    
+    def fetch_contents(current_path):
+        url = f"https://api.github.com/repos/{owner}/{repo}/contents/{current_path}"
+        response = requests.get(url)
+        contents = response.json()
+        
+        # Handle single file response
+        if not isinstance(contents, list):
+            contents = [contents]
+
+        msg = contents[0].get("message") 
+        if msg and msg.startswith("API rate limit exceeded"):
+                logger.error("GitHub API limit exceeded")
```

Comment:
Ah - when this happens do you remember how to fix it? Can you add a line of docs or a hint in this error?

Not critical, but it might make someone's day in the future